### DISCUSSION TITLE
KongConfigurationTranslationFailed

### DISCUSSION THREAD

**Discussion by balait4** [Kong/kubernetes-ingress-controller#6489] [General]
Hi, After upgrade from 3.5.0 to 3.7.1 and KIC 3.0 to 3.2.3, I'm seeing the below issue.
This is dbless and I upgraded manually, I applied the roles and crd from the below link. 
But still its not working, seems something missing in roles, could thought/help on this please?

kubectl kustomize https://github.com/Kong/kubernetes-ingress-controller/config/crd?ref=v3.2.3 | kubectl apply -f -
kubectl kustomize https://github.com/Kong/kubernetes-ingress-controller/config/rbac?ref=v3.2.3 | kubectl apply -f -

I also installed the Install the Gateway APIs as per the https://docs.konghq.com/kubernetes-ingress-controller/latest/install/helm/#installation for first time as we dont have that before.
```
 2024-09-06T13:13:35Z error recording a Warning event for object {"name": "northamerica-lb-header", "namespace": "api-proxy", "kind": "Service", "apiVersion": "/v1", "reason": "KongConfigurationTranslationFailed", "message": "failed fetching KongUpstreamPolicy: KongUpstreamPolicy api-proxy/kongingress-lb-header not found", "error": "object failed to apply"}
 2024-09-06T13:13:35Z error recording a Warning event for object {"name": "britishisles-least-conn", "namespace": "api-proxy", "kind": "Service", "apiVersion": "/v1", "reason": "KongConfigurationTranslationFailed", "message": "failed fetching KongUpstreamPolicy: KongUpstreamPolicy api-proxy/kongingress-least-conn not found", "error": "object failed to apply"}
 2024-09-06T13:13:35Z error recording a Warning event for object {"name": "netherlands-funda-lb-header", "namespace": "api-proxy", "kind": "Service", "apiVersion": "/v1", "reason": "KongConfigurationTranslationFailed", "message": "failed fetching KongUpstreamPolicy: KongUpstreamPolicy api-proxy/kongingress-lb-header not found", "error": "object failed to apply"}
 2024-09-06T13:13:35Z error recording a Warning event for object {"name": "westcentraleurope-lb-header", "namespace": "api-proxy", "kind": "Service", "apiVersion": "/v1", "reason": "KongConfigurationTranslationFailed", "message": "failed fetching KongUpstreamPolicy: KongUpstreamPolicy api-proxy/kongingress-lb-header not found", "error": "object failed to apply"}
```

**Comment by balait4** [comment]
@pmalek Any help on this please? I tried applying all the roles and rbac and crds from the KIC v3.2.3, still not working. I tried uninstalling the newly setup Gateway APIs still no luck. But it looks like some roles are missing. Thanks

**Reply by pmalek** [reply]
Please post the manifests that you've used which got you these errors.

**Reply by balait4** [reply]
Thanks for the reply! Here is the KUP manifest, similarly we have for least-conn and lb-header
[KongUpstreamPolicy.txt](https://github.com/user-attachments/files/16981975/KongUpstreamPolicy.txt)

**Reply by pmalek** [reply]
Can you also provide the `Service` manifest(s)?

**Reply by balait4** [reply]
Here is the service manifest
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    konghq.com/retries: '2'
    konghq.com/upstream-policy: kongingress-apikey-header
    konghq.com/plugins: request-transformer-plugin
#    konghq.com/plugins: request-transformer-plugin, prevent-retry-on-timeout-plugin
    konghq.com/protocol: https
    konghq.com/connect-timeout: "2000"
    konghq.com/read-timeout: "300000"
  name: australia-apikey-header
  namespace: api-proxy
spec:
  clusterIP: None
  ports:
  - name: apps
    port: 8080
    protocol: TCP
    targetPort: 8080
  type: ClusterIP
  ```

**Reply by balait4** [reply]
Any other role/crd we need to install on top of what I installed? 
kubectl kustomize https://github.com/Kong/kubernetes-ingress-controller/config/crd?ref=v3.2.3 | kubectl apply -f -
kubectl kustomize https://github.com/Kong/kubernetes-ingress-controller/config/rbac?ref=v3.2.3 | kubectl apply -f -

**Reply by pmalek** [reply]
I'm having hard time understanding what is the problem here:

The original post contains logs mentioning `KongUpstreamPolicy` `api-proxy/kongingress-lb-header`. The provided txt file contains a dump of `KongUpstreamPolicy` `kongingress-apikey-header`.

The manifest above described the `Service` `australia-apikey-header` whereas the original message had logs mentioned different `Services`.

If you could please put all those information (relevant for this issue) into one post that would help immensely.

---

As for the kustomize manifests and roles: our supported way of installation is through https://github.com/Kong/charts/. All charts there contain the relevant roles.

Having said that, what's in the kustomize manifests should also suffice. The error would indicate that the problem is not with permissions but rather KIC not being able to find the `KongUpstreamPolicy`.

**Reply by balait4** [reply]
Hi,
Here is the one complete service manifest, we have multiple service which uses the KongUpstreamPolicy, thats why it got mixed in my previous attachement. 
[KongUpstreamPolicy.txt](https://github.com/user-attachments/files/17028307/KongUpstreamPolicy.txt)

**Reply by balait4** [reply]
I tried upgrade to KIC version v3.3.1 where still I'm facing the same issue. Currently we are at KIC v3.0.2(where it is working fine). Here is the attachment for my kong deployment manifests and KIC starting logs where any clue what wrong with this KIC?
[deployment.txt](https://github.com/user-attachments/files/17034394/deployment.txt)

**Reply by balait4** [reply]
Is there any detailed log or debug to see why KIC not picking the `KongUpstreamPolicy`?

**Reply by pmalek** [reply]
You can try inspecting the `KongUpstreamPolicy` object status. Its conditions might indicate what's wrong. Additionally you might enable debug logs and search for anything relevant for this `KongUpstreamPolicy`.

Feel free to share the object status and logs here.

**Reply by balait4** [reply]
thanks for the reply! Here is the log of `KongUpstreamPolicy` object status. And KIC debug logs. 
[KUP_Object_status.txt](https://github.com/user-attachments/files/17144161/KUP_Object_status.txt)
[debug_log.txt](https://github.com/user-attachments/files/17144163/debug_log.txt)

**Reply by balait4** [reply]
Hi, One more thing, few KUP is working fine and seems only few are failing but there is no difference in the yaml, here is the working and non-working KUP object.
[KUP_non_working_object.txt](https://github.com/user-attachments/files/17147724/KUP_non_working_object.txt)
[KUP_working_object_status.txt](https://github.com/user-attachments/files/17147726/KUP_working_object_status.txt)

Non-working KUP resource status block is continuously updated with ancestor reordering.

![image](https://github.com/user-attachments/assets/89eae5a7-b2a9-4c82-8452-a1e5c8ea935c)

**Reply by balait4** [reply]
Also attaching the debug log from kong and KIC, I have issue only for the KUP from the namespace api-proxy. KUP from other namespace seems to be working. 
[debug_log_KIC.txt](https://github.com/user-attachments/files/17152754/debug_log_KIC.txt)
[debug_log_kong.txt](https://github.com/user-attachments/files/17152757/debug_log_kong.txt)

**Comment by balait4** [comment]
Any help on this please? As I'm not able to move from version v3.0.2. I want to upgrade to latest version to use the env variable `CONTROLLER_GATEWAY_TO_RECONCILE`.
